### PR TITLE
Use a centralized thread pool for all of Amplify

### PR DESCRIPTION
**Pull request by jamesonwilliams**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java [hunk 1]
```diff
@@ -162,7 +162,7 @@ public synchronized void initialize(
         Objects.requireNonNull(onSuccess);
         Objects.requireNonNull(onError);
 
-        threadPool.submit(() -> {
+        threadPool.execute(() -> {
```
Comment (raphkim) [inline aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java]: Nice catch. TIL the difference between the two.

File: core/src/main/java/com/amplifyframework/core/async/AmplifyExecutors.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private static final String THREAD_GROUP_NAME = "Amplify";
+    private static final ThreadGroup THREAD_GROUP = threadGroup();
+    private static final ExecutorService STANDARD = standardExecutorService();
+    private static final ScheduledExecutorService PERIODIC = scheduledExecutorService();
+
+    @SuppressWarnings("checkstyle:all") private AmplifyExecutors() {}
+
+    /**
+     * Gets a fixed thread pool {@link ExecutorService} with as many threads
+     * as there are processors on the host.
```
Comment (raphkim) [inline core/src/main/java/com/amplifyframework/core/async/AmplifyExecutors.java]: Maybe have another one where we can set pool size?

File: core/src/main/java/com/amplifyframework/core/Amplify.java [hunk 3]
```diff
@@ -74,7 +74,7 @@
     private static final AtomicBoolean CONFIGURATION_LOCK = new AtomicBoolean(false);
 
     // An executor on which categories may be initialized.
-    private static final ExecutorService INITIALIZATION_POOL = Executors.newFixedThreadPool(CATEGORIES.size());
```
Comment (raphkim) [inline core/src/main/java/com/amplifyframework/core/Amplify.java]: I think it makes sense to keep this logic? What if the host processor only has 1 core (hypothetically speaking), and it began initializing datastore first? All other categories will be blocked until datastore is initialized right?

**Comment by jamesonwilliams** [comment]
This PR has gone stale. Let's re-open it, if/when it will receive an update. It is still here for reference, should we wish to use it for that purpose, meanwhile.

**Pull request closed without merging**